### ISSUE TITLE
Highlight corresponding source code when step is selected, and the other way around

### ISSUE THREAD

**Issue by igarashitm** [KaotoIO/kaoto-ui#1431]
### Please describe the feature that you want to propose

When a step is selected in the canvas, highlight the corresponding lines in the source code editor.

Also when the code range is selected, highlight corresponding step(s).

![Screenshot from 2023-03-09 09-28-57](https://user-images.githubusercontent.com/265462/224057103-5cbd0ad4-2371-4df7-9735-83a5b29b8f21.png)

**Comment by kahboom** [comment]
This would be really cool, but unfortunately only possible if we decide to start parsing the YAML. That was something we didn't want to do from the beginning, maybe @Delawen can also shed some light here as it relates to multi-DSL.

**Comment by igarashitm** [comment]
Yes, this would need several big jumps, parsing YAML would be monaco-editor + LSP and more tight integration with source code editor, and then UI would need to be highly pluggable, if we want to stay DSL agnostic on ui core code.
